### PR TITLE
ddl: fix error when setting `tidb_enable_ddl`

### LINKED ISSUE #41277: set tidb_enable_ddl=false, but still can create table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
mysql> show config where name like '%ddl%';
+------+-------------------+-----------------------------+-------+
| Type | Instance          | Name                        | Value |
+------+-------------------+-----------------------------+-------+
| tidb | 172.16.5.101:4000 | instance.ddl_slow_threshold | 300   |
| tidb | 172.16.5.101:4000 | instance.tidb_enable_ddl    | true  |
+------+-------------------+-----------------------------+-------+
2 rows in set (0.02 sec)

mysql> set tidb_enable_ddl=false;
ERROR 8246 (HY000): Error happened when enable/disable DDL: can not disable ddl when there is only one instance
mysql> set tidb_enable_ddl=false;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------------------------------------------+
| Level   | Code | Message                                                                       |
+---------+------+-------------------------------------------------------------------------------+
| Warning | 8142 | modifying tidb_enable_ddl will require SET GLOBAL in a future version of TiDB |
+---------+------+-------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> show config where name like '%ddl%';
+------+-------------------+-----------------------------+-------+
| Type | Instance          | Name                        | Value |
+------+-------------------+-----------------------------+-------+
| tidb | 172.16.5.101:4000 | instance.ddl_slow_threshold | 300   |
| tidb | 172.16.5.101:4000 | instance.tidb_enable_ddl    | false |
+------+-------------------+-----------------------------+-------+
2 rows in set (0.02 sec)

mysql> use test;
Database changed
mysql> create table t1(a int);
Query OK, 0 rows affected (1.09 sec)

```

### 2. What did you expect to see? (Required)
create table fail

### 3. What did you see instead (Required)
create table success

### 4. What is your TiDB version? (Required)

[tidb@101 ~]$ /home/tidb/deploy/tidb-4000/bin/tidb-server -V
Release Version: v6.5.0
Edition: Community
Git Commit Hash: 706c3fa3c526cdba5b3e9f066b1a568fb96c56e3
Git Branch: heads/refs/tags/v6.5.0
UTC Build Time: 2022-12-27 03:50:44
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore
[tidb@101 ~]$

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #41277 

Problem Summary:

In `SetGlobal` of `tidb_enable_ddl`, when `switchDDL` returns error, the modification of `TiDBEnableDDL` should be skipped.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix an error in setting `tidb_enable_ddl` to `OFF` when there is only one tidb instance.
```

### CODE REVIEW

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -4386,3 +4386,14 @@ PARTITION BY LIST (level) (
 	tk.MustExec(`INSERT INTO member_level (id, level) values (313, 6)`)
 	tk.MustContainErrMsg(`ALTER TABLE member_level REORGANIZE PARTITION lEven INTO (PARTITION lEven VALUES IN (2,4))`, "[table:1526]Table has no partition for value 6")
 }
+
+func TestIssue41277(t *testing.T) {
```
Comment (CbcWestwolf) [inline ddl/db_integration_test.go:4390]:
```suggestion
func TestDisableDDL(t *testing.T) {
	// https://github.com/pingcap/tidb/issues/41277
```

**Review by hawkingrei — APPROVED** [review]

File: ddl/db_integration_test.go [hunk 2]
```diff
@@ -4386,3 +4386,15 @@ PARTITION BY LIST (level) (
...
+
+func TestDisableDDL(t *testing.T) {
+	// https://github.com/pingcap/tidb/issues/41277
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	tk.MustQuery("select @@global.tidb_enable_ddl").Check(testkit.Rows("1"))
+	tk.MustGetErrCode("set @@global.tidb_enable_ddl=false;", errno.ErrDDLSetting)
+	tk.MustGetErrCode("set @@global.tidb_enable_ddl=false;", errno.ErrDDLSetting)
```
Comment (Defined2014) [inline ddl/db_integration_test.go:4398]: One line is enough.

**Review by Defined2014 — APPROVED** [review]
Reset LGTM

**Pull request merged**